core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -64,6 +64,12 @@ private[spark] class Executor(
 
   logInfo(s"Starting executor ID $executorId on host $executorHostname")
 
+  private var executorShutdown = false
+  ShutdownHookManager.addShutdownHook(
+    () => if (!executorShutdown) {
+      pluginShutdown
```

Comment:
Why single out plugin shutdown?

This should just call `stop()`. And `stop()` should be idempotent.